extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
(8) will not happen, because at that point the executor thread will be in the loop over `for (... : kafkaStream)` which is an infinite loop and does not contain any checking for shutdown.

So I think that loop needs to check for the executor service being stopped. I also think a `shutdownNow` is still called for, since the `kafkaStream` iterator may block forever if no new messages are forthcoming in the Kafka topic, and that needs to be dealt with. It's probably a likely situation for a lookup oriented topic to be in, since I bet new messages may not come very often.